libs/SmartSync/SmartSync/Classes/Target/SFSyncUpTarget.m

```diff
@@ -0,0 +1,281 @@
+        self.isDeleted = isDeleted;
+    }
+    return self;
+}
+
+@end
+
+typedef void (^SFSyncUpRecordModDateBlock)(SFRecordModDate *remoteModDate);
+
+@interface  SFSyncUpTarget ()
+@property (nonatomic, strong) NSArray*  createFieldlist;
+@property (nonatomic, strong) NSArray*  updateFieldlist;
+@end
+
+@implementation SFSyncUpTarget
+
+#pragma mark - Initialization and serialization methods
+
+- (instancetype)initWithDict:(NSDictionary *)dict {
+    return [self initWithCreateFieldlist:dict[kSFSyncUpTargetCreateFieldlist] updateFieldlist:dict[kSFSyncUpTargetUpdateFieldlist]];
+}
+
+- (instancetype)init {
+    return [self initWithCreateFieldlist:nil updateFieldlist:nil];
+}
+
+- (instancetype)initWithCreateFieldlist:(NSArray *)createFieldlist
+                        updateFieldlist:(NSArray *)updateFieldlist {
+    self = [super init];
+    if (self) {
+        self.targetType = SFSyncUpTargetTypeRestStandard;
+        self.createFieldlist = createFieldlist;
+        self.updateFieldlist = updateFieldlist;
+    }
+    return self;
+}
+
+
++ (instancetype)newFromDict:(NSDictionary*)dict {
+    // We should have an implementation class or a target type
+    NSString* implClassName = dict[kSFSyncTargetiOSImplKey];
+    if (implClassName.length > 0) {
+        Class customSyncUpClass = NSClassFromString(implClassName);
+        if (![customSyncUpClass isSubclassOfClass:[SFSyncUpTarget class]]) {
+            [SFLogger log:self level:SFLogLevelError format:@"%@ Class '%@' is not a subclass of %@.", NSStringFromSelector(_cmd), implClassName, NSStringFromClass([SFSyncUpTarget class])];
+            return nil;
+        } else {
+            return [[customSyncUpClass alloc] initWithDict:dict];
+        }        
+    }
+    // No implementation class - using target type
+    else {
+        // No target type - assume kSFSyncUpTargetTypeRestStandard (hybrid apps don't specify it a sync up target type by default)
+        NSString *targetTypeString = (dict[kSFSyncTargetTypeKey] == nil ? kSFSyncUpTargetTypeRestStandard : dict[kSFSyncTargetTypeKey]);
+        switch ([self targetTypeFromString:targetTypeString]) {
+            case SFSyncUpTargetTypeRestStandard:
+                return [[SFSyncUpTarget alloc] initWithDict:dict];
+                
+            case SFSyncUpTargetTypeCustom:
+                [SFLogger log:self level:SFLogLevelError format:@"%@ Custom class name not specified.", NSStringFromSelector(_cmd)];
+                return nil;
+        }
+    }
+}
+
+- (NSMutableDictionary *)asDict {
+    NSMutableDictionary *dict = [super asDict];
+    dict[kSFSyncTargetTypeKey] = [[self class] targetTypeToString:self.targetType];
+    if (self.createFieldlist) dict[kSFSyncUpTargetCreateFieldlist] = self.createFieldlist;
+    if (self.updateFieldlist) dict[kSFSyncUpTargetUpdateFieldlist] = self.updateFieldlist;
+    return dict;
+}
+
+# pragma mark - Public sync up methods
+
+- (void)isNewerThanServer:(SFSmartSyncSyncManager *)syncManager
```

Comment:
Smarter than before.
Just like on Android, we deal with locally deleted and/or locally deleted records better.